### PR TITLE
WIP Bug 479659 - [wizard] User feedback has potential for improvements

### PR DESCRIPTION
**Pull request by dhuebner**
Signed-off-by: Dennis Huebner <dennis.huebner@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=479659

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend [hunk 1]
```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
...
+		if (createUiProject.selection && sourceLayout.selectionIndex != 0) {
+			if (createUiProject === source) {
+				reportIssue(ERROR, '''
+				'�createUiProject.text�' requier Flat source layout.
+				Please <a>select 'Flat'</a> source layout.''', [
+					sourceLayout.select(0)
+				])
+			} else {
+				reportIssue(ERROR, '''
+				Flat source layout is not supported bei the '�createUiProject.text�' project.
```
Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: Bei =)

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend [hunk 2]
```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
...
+		pageComplete = statusWidget.severtity !== ERROR
+	}
+
+	def checkWidgets(SelectionEvent e) {
+		val source = e?.source
+		if (createUiProject.selection && sourceLayout.selectionIndex != 0) {
+			if (createUiProject === source) {
+				reportIssue(ERROR, '''
+				'�createUiProject.text�' requier Flat source layout.
+				Please <a>select 'Flat'</a> source layout.''', [
```
Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: Why different messages depending on what the user clicked first?

Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: Ah, you offer "quick fixes". In that case, why don't you always offer both solutions?

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend [hunk 3]
```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
...
+			if (preferredBuildSystem === source) {
+				reportIssue(ERROR, '''
+				The '�createWebProject.text�' project can not be build using Eclipse-PDE build.
+				Please <a>deselect '�createWebProject.text�'</a>.''', [
+					createWebProject.selection = false
+				])
+			} else {
+				reportIssue(WARNING, '''
+				To build the '�createWebProject.text�' project, you need to choose maven or gradle build system.
+				Select <a>maven</a> build.''', [
```
Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: "Select Gradle" as a second solution would be nice.

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend [hunk 4]
```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
...
+				])
+			}
+		}
+
+		val dependend = #[createUiProject, createIdeaProject, createWebProject]
+		if (!createIdeProject.selection && dependend.exists[selection]) {
+			val affectedProjects = dependend.filter[selection].join(', ', [text])
+			if (createIdeProject === source) {
+				reportIssue(ERROR, '''
+				Projects like '�affectedProjects�' depends on '�createIdeProject.text�' project.
```
Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: 'Frontend projects'

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend [hunk 5]
```diff
@@ -73,18 +87,115 @@ class AdvancedNewProjectPage extends WizardPage {
...
+		checkWidgets(e)
+		pageComplete = statusWidget.severtity !== ERROR
+	}
+
+	def checkWidgets(SelectionEvent e) {
+		val source = e?.source
+		if (createUiProject.selection && sourceLayout.selectionIndex != 0) {
+			if (createUiProject === source) {
+				reportIssue(ERROR, '''
+				'�createUiProject.text�' requier Flat source layout.
```
Comment (svenefftinge) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: '... requires a flat source layout'

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend (hunk 2 above)
Comment (svenefftinge) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: 'Plain' not 'Flat'

File: plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend (hunk 1 above)
Comment (svenefftinge) [inline plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.xtend]: Plain

**Pull request merged**